protocol/src/protocol.ts

```diff
@@ -502,6 +502,11 @@ export interface _ClientCapabilities {
 	 */
 	textDocument?: TextDocumentClientCapabilities;
 
+	/**
+	 * Whether the client supports diagnostics with related information.
+	 */
+	diagnosticRelatedInformation?: boolean;
```

Comment:
I don't think that this property belongs here. Since diagnostics are in the form of a `textDocument/publishDiagnostics` request, perhaps it should go there instead?

Reply:
Makes sense. I moved the capability. Thanks!